### ISSUE TITLE
IMGUI_API ImPool missing symbols with Implot

### ISSUE THREAD

**Issue by jvannugteren** [ocornut/imgui#4962]
This is quite an interesting issue I discovered. It relates to implot, imgui and possibly magnum-integration, where we discussed this issue as well: https://github.com/mosra/magnum-integration/issues/94. 

The problem arises only on Windows with msvc, when compiling `implot.cpp` (as part of shared library), which uses the `ImPool` template class from `imgui_integration.h`. This results in several missing symbol errors, similar to this:
```
error LNK2019: unresolved external symbol "__declspec(dllimport) public: struct ImPlotItem * __cdecl ImPool<struct ImPlotItem>::GetOrAddByKey(unsigned int)" (__imp_?GetOrAddByKey@?$ImPool@UImPlotItem@@@@QEAAPEAUImPlotItem@@I@Z) referenced in function "public: struct ImPlotItem * __cdecl ImPlotItemGroup::GetOrAddItem(unsigned int)" (?GetOrAddItem@ImPlotItemGroup@@QEAAPEAUImPlotItem@@I@Z)
```

The issue can be resolved by changing the ImPool class definition from `template<typename T> struct IMGUI_API ImPool` to `template<typename T> struct __declspec( dllexport ) ImPool` or to `template<typename T> struct ImPool`. It seems to us that the dll export is incompatible with a template class and shouldn't be there (see comment of pezcode in the link above). 

In the process I also created a minimalist example to reproduce this error:
https://gitlab.com/jeroen.van.nugteren/ImPoolProblem

Thank you for looking at this.

**Comment by ocornut** [comment]
Thank you for reporting!

They are inline by definition so IMGUI_API isn't needed anyway.
Pushed a fix for it now: 64d910a

**Comment by jvannugteren** [comment]
Thanks. I think that fixes it.
